File: 프로그래머스/0/120852. 소인수분해/소인수분해.py
def solution(n):
    answer = set()
    
    d=2

    while n!=1:            
        while n%d==0:
            n=n//d  
            answer.add(d)
        d=d+1
        
    if n > 1:
        answer.add(n)
    a = sorted(list(answer))
    # list(answer).sort()
    return a

File: 프로그래머스/0/120852. 소인수분해/소인수분해.java
import java.util.*;

class Solution {
    public int[] solution(int n) {
        List<Integer> list = new ArrayList<>();
        
        int d = 2;
        while (n > 1) {
            if (n % d == 0) {
                if (!list.contains(d)) { 
                    list.add(d);
                }
                n /= d;
            } else {
                d++;
            }
        }

        int[] answer = new int[list.size()];
        for (int i = 0; i < list.size(); i++) {
            answer[i] = list.get(i);
        }

        return answer;
    }
}